src/Grpc/test/testassets/InteropWebsite/Directory.Build.targets

```diff
@@ -0,0 +1,24 @@
+<Project>
+  <!-- Skip the parent folder to prevent getting test package references. -->
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory)..\..\, Directory.Build.targets))\Directory.Build.targets" />
+
+  <PropertyGroup>
+    <RestoreAdditionalProjectSources>$(RestoreAdditionalProjectSources);$(ArtifactsShippingPackagesDir)</RestoreAdditionalProjectSources>
+    <MicrosoftAspNetCoreAppRefPackageVersion Condition="'$(IsTargetingPackBuilding)' != 'false'">$(TargetingPackVersion)</MicrosoftAspNetCoreAppRefPackageVersion>
+    <MicrosoftAspNetCoreAppRefPackageVersion Condition="'$(IsTargetingPackBuilding)' == 'false'">$(AspNetCoreBaselineVersion)</MicrosoftAspNetCoreAppRefPackageVersion>
+  </PropertyGroup>
+
+  <!-- Use the shared framework that was produced -->
+  <ItemGroup>
+    <KnownFrameworkReference
+      Include="Microsoft.AspNetCore.App"
+      TargetFramework="$(DefaultNetCoreTargetFramework)"
+      RuntimeFrameworkName="Microsoft.AspNetCore.App"
+      TargetingPackName="Microsoft.AspNetCore.App.Ref"
+      RuntimePackNamePatterns="Microsoft.AspNetCore.App.Runtime.**RID**"
+      DefaultRuntimeFrameworkVersion="$(SharedFxVersion)"
+      LatestRuntimeFrameworkVersion="$(SharedFxVersion)"
+      TargetingPackVersion="$(MicrosoftAspNetCoreAppRefPackageVersion)"
```

Comment:
Doesn't work unless the targeting pack is building.

This is a fundamental break that could be left out completely because the `$(TestDependsOnAspNetRuntime)` infrastructure handles everything for you on the Helix agents. If the point is to handle things on build agents, recommend using `$(UpdateAspNetCoreKnownFramework)` property but ignoring the test asset projects except in very special cases.